instance/tasks.py

```diff
@@ -339,3 +344,111 @@ def cleanup_old_betatest_users():
         users_to_delete = queryset.filter(is_active=False, profile__modified__lte=delete_cutoff)
         logger.info('Deleting %s users.', users_to_delete.count())
         users_to_delete.delete()
+
+
+class KillZombiesRunner:
+    """
+    Helper class to run `kill_zombies_periodically`
+    """
+
+    def __init__(self):
+        self.region: str = getattr(settings, "OPENSTACK_REGION", "gra")
+        self.recipient: list = [email for _, email in getattr(settings, "ADMINS", [])]
+        self.threshold: int = getattr(settings, "KILL_ZOMBIES_WARNING_THRESHOLD")
+
+    def send_email(self, subject: str, body: str) -> int:
+        """
+        Utility method for sending emails
+        """
+        if self.recipient:
+            email = EmailMessage(
+                subject, body, settings.DEFAULT_FROM_EMAIL, self.recipient
+            )
+            return email.send()
+        else:
+            logging.info("Email recipient is undefined. Skipping email.")
+            return None
+
+    def trigger_warning(self):
+        """
+        Warns when more than `threshold` VMs will be terminated
+        """
+        subject = "{} Zombie instances over current threshold ({})".format(
+            datetime.now().strftime("%Y-%m-d"), self.threshold
+        )
+        body = (
+            "The number of instances in region {} to be terminated"
+            "is over KILL_ZOMBIES_WARNING_THRESHOLD ({}).\n"
+        ).format(self.region, self.threshold)
+        logging.info(
+            "%s\nSending an email notification to: %s",
+            body, self.recipient
+        )
+        self.send_email(subject, body)
+
+    def handle_dry_run_output(self, stdout: str) -> int:
+        """
+        If there are servers to terminate, a `kill_zombies` dry run will warn:
+        "Would have terminated {} zombies" to stdout.
+        """
+        if "No servers found in region" in stdout:
+            return 0
+        try:
+            return int(stdout.split("Would have terminated")[1].split()[0])
+        except IndexError:
+            logger.info("Received unexpected input from dry run")
+            return None
+
+
+if getattr(settings, "KILL_ZOMBIES_ENABLED"):
+    @db_periodic_task(crontab(
+        minute=KILL_ZOMBIES_CRON_SCHEDULE[0],
+        hour=KILL_ZOMBIES_CRON_SCHEDULE[1],
+        day=KILL_ZOMBIES_CRON_SCHEDULE[2],
+        month=KILL_ZOMBIES_CRON_SCHEDULE[3],
+        day_of_week=KILL_ZOMBIES_CRON_SCHEDULE[4],
+    ))
+    def kill_zombies_periodically_task():
```

Comment:
@spokerman12, the `periodically` part is already implied by the usage of the `db_periodic_task` decorator above and imho, it shouldn't be added in the name.

Also, why is this wrapper needed in the first place instead of adding the implementation directly within this function?

Reply:
This is a design oversight caused by side effects of addressing the first review. I'll implement the request.

Reply:
The wrapper is necessary because if a periodic task is defined, Huey will run it unless we explicitly tell it not to.

Reply:
@spokerman12, I don't understand your answer. Why can't the body of the `kill_zombies_periodically()` function be directly within this `kill_zombies_periodically_task()` function? That is the wrapping that I asked about in my previous comment.

Reply:
OK I see what you mean now; I didn't see it from the `run` method being an addition from the reviews.
I'll apply this.